src/uicomponents/LinkButton.tsx

```diff
@@ -14,10 +14,44 @@ import { RaidBattleInfo } from "../raidcalc/RaidBattle";
 
 import PokedexService from "../services/getdata";
 import { MoveData, RaidTurnInfo, SubstituteBuildInfo, TurnGroupInfo } from "../raidcalc/interface";
-import { getTranslation } from "../utils";
+import { encode, decode, getTranslation } from "../utils";
 
+import { db } from "../config/firestore";
+import { collection, doc, getCountFromServer, getDoc, setDoc } from "firebase/firestore";
+import STRAT_LIST from "../data/strats/stratlist.json";
 
 const gen = Generations.get(9);
+const JSON_HASHES = Object.entries(STRAT_LIST).map(([boss, strats]) => Object.entries(strats as Object).map(([name, h]) => h as string)).flat();
+
+async function getFullHashFromShortHash(hash: string): Promise<string> {
+    let cleanHash = hash;
+    if (hash[0] === "#") {
+        cleanHash = hash.slice(1);
+    }
+    const hashID = decode(cleanHash);
+    const docRef = doc(db, "links", `${hashID}`);
+    const docSnap = await getDoc(docRef);
+    if (!docSnap.exists()) {
+      console.log("Invalid link");
+    }
+    const data = docSnap.data();
+    return data ? data.hash : null;
+}
+
+async function generateShortHash(): Promise<[number, string]> {
+    const coll = collection(db, "links");
+    const snapshot = await getCountFromServer(coll);
+    const dbLen = snapshot.data().count;
+    const shortHash = encode(dbLen);
+    return [dbLen, shortHash]
+}
+
+async function setLinkDocument(id: number, shortHash: string, fullHash: string): Promise<void> {
+    await setDoc(doc(db, "links", `${id}`), {
```

Comment:
Currently the serialized data is being stored in the database which would make it difficult to perform queries on. Is there any info contained in a strat that might be useful to store raw for querying? (strat names, Pokemon used, moves used, credits, etc). I dont have anything in mind at the moment, just bringing up the consideration.

Reply:
That's a good thought. It would be really cool to be able to search the database for particular strats. I'm not sure how exactly database reads are calculated by Firestore Cloud, so I'm a bit worried that such a feature could make read usage skyrocket.

It won't hurt our storage space much to include a few more details, though. I think strat name, boss species, and raider species are probably most relevant.